decoder.rb

```diff
@@ -0,0 +1,42 @@
+MORSE_CODE_CONTAINER = {
+  'A' => '.-', 'B' => '-...',
+  'C' => '-.-.', 'D' => '-..', 'E' => '.',
+  'F' => '..-.', 'G' => '--.', 'H' => '....',
+  'I' => '..', 'J' => '.---', 'K' => '-.-',
+  'L' => '.-..', 'M' => '--', 'N' => '-.',
+  'O' => '---', 'P' => '.--.', 'Q' => '--.-',
+  'R' => '.-.', 'S' => '...', 'T' => '-',
+  'U' => '..-', 'V' => '...-', 'W' => '.--',
+  'X' => '-..-', 'Y' => '-.--', 'Z' => '--..',
+  '1' => '.----', '2' => '..---', '3' => '...--',
+  '4' => '....-', '5' => '.....', '6' => '-....',
+  '7' => '--...', '8' => '---..', '9' => '----.',
+  '0' => '-----', ', ' => '--..--', '.' => '.-.-.-',
+  '?' => '..--..', '/' => '-..-.', '-' => '-....-',
+  '(' => '-.--.', ')' => '-.--.-'
+}.freeze
+
+def decode_char(mo_char)
+  char_decoded = ''
+  MORSE_CODE_CONTAINER.each do |key, value|
+    char_decoded += key if mo_char == value
+  end
+  char_decoded
+end
+
+def decode_word(mo_word)
+  word_decoded = ''
+  mo_word = mo_word.split
+  mo_word.each { |char| word_decoded += decode_char(char) }
+  word_decoded
+end
+
+def decode_message(mo_message)
+  message_decoded = ''
+  mo_message = mo_message.split('   ')
+  mo_message.each { |msg| message_decoded += " #{decode_word(msg)}" }
+  print(message_decoded.strip)
```

Comment:
- [ ] Remove the print just only return the decoded message because I am getting a `nil` at the end of the decoded message check here :point_down: 

![image](https://user-images.githubusercontent.com/65068771/156572587-2c940263-bf2f-467d-be1f-890f082a20ad.png)